webapp/app/common/emergency-proposal-schema.ts

```diff
@@ -0,0 +1,45 @@
+import { getAddress, isAddress, padHex } from "viem";
+import { z } from "zod";
+
+export const basicPropSchema = z.object({
+  targetAddress: z.string().refine((value) => isAddress(value), {
+    message: "Invalid Ethereum address",
+  }),
+  calldata: z
+    .string()
+    .regex(/^0x[a-fA-F0-9]*$/, "Calldata must be a hex string starting with 0x")
+    .refine((value) => value.length % 2 === 0, {
+      message: "Calldata must be valid hex-encoded bytes",
+    }),
+  value: z
+    .string()
+    .regex(/^\d*\.?\d*$/, "Value must be a positive number")
```

Comment:
This can be changed to z.coerce.number()

Reply:
The problem with using coerce is that it transform  this to a number. And everything is made to have a string here. I'm changing this in my next PR, so it's going to be improved.